app/queries/facility_yearly_follow_ups_and_registrations_query.rb

```diff
@@ -0,0 +1,58 @@
+class FacilityYearlyFollowUpsAndRegistrationsQuery
+  NON_COUNT_FIELDS = %i[
+    block_region_id
+    district_region_id
+    facility_id
+    facility_region_id
+    facility_region_slug
+    month_date
+    state_region_id
+  ]
+  FINANCIAL_YEAR_START_MONTH = 4
+  YEAR_START_MONTH = 1
+
+  def initialize(region, current_user)
+    @region = region
+    @current_user = current_user
+    @year_start_month = Flipper.enabled?(:progress_financial_year, @current_user) ? FINANCIAL_YEAR_START_MONTH : YEAR_START_MONTH
```

Comment:
Consider a name that's agnostic of touchpoint in the app (remove "progress"), and very specific about april and march and stuff.

✨ `yearly_reports_start_from_april` ✨ 
`yearly_reports_start_in_april`
`yearly_reports_from_april_to_march`
`yearly_reports_from_april`
`...`